src/equisolve/numpy/utils.py

```diff
@@ -35,92 +32,15 @@ def block_to_array(block: TensorBlock, parameter_keys: List[str]) -> np.ndarray:
     work for Kernel models.
     """
     M = []
-    for parameter in parameter_keys:
+    for parameter in ["values"] + block.gradients_list():
         if parameter == "values":
-            data = block.values
+            values = block.values
         else:
-            data = block.gradient(parameter).values
-        M.append(data.reshape(np.prod(data.shape[:-1]), data.shape[-1]))
+            values = block.gradient(parameter).values
+        M.append(values.reshape(np.prod(values.shape[:-1]), values.shape[-1]))
```

Comment:
suggestion is a bit broken. -1 just takes the remaining dimensions
```python
    M = [block.values.reshape(-1, block.values.shape[-1])]
    for parameter in block.gradients_list():
        values= block.gradient(parameter).values
        M.append(values.reshape(-1, values.shape[-1]))
``

Reply:
off very nice suggestion!